server/sonar-server/src/main/java/org/sonar/server/es/EsClientProvider.java

```diff
@@ -56,13 +56,14 @@ public EsClient provide(Settings settings) {
         esSettings.put("client.transport.sniff", true);
         nativeClient = TransportClient.builder().settings(esSettings).build();
         Arrays.stream(settings.getStringArray(ProcessProperties.CLUSTER_SEARCH_HOSTS))
-          .map(Host::parse)
-          .forEach(h -> h.addTo(nativeClient));
+          .map(HostAndPort::fromString)
```

Comment:
I don't think it's inconsistency there but this reflects the two modes to connect to ElasticSearch in a cluster environment:

- If Elasticsearch is activated on the node, SonarQube is not trying to connect to Elasticsearch as a cluster but connect to the local Elasticsearch node (so using `SEARCH_HOST` ie. the interface bound and `SEARCH_PORT`, and I think 0 should never be used in production mode)
- If Elasticsearch is not activated on the node, SonarQube is using `CLUSTER_SEARCH_HOSTS` that contains the list of Elasticsearch nodes (with or without port number). And in this case omitting port number should not lead to an exception but using the default Elasticsearch port.